micro-rdk-nmea/src/parse_helpers/parsers.rs

```diff
@@ -0,0 +1,362 @@
+                    let type_size = std::mem::size_of::<Self::FieldType>();
+                    match self.bit_size {
+                        x if x == (type_size * 8) => {
+                            let value: &[u8] = &data[start_idx..(start_idx + type_size)];
+                            Ok((start_idx + type_size, <$t>::from_le_bytes(value.try_into()?)))
+                        }
+                        x => {
+                            let end_idx = start_idx + (x / 8);
+
+                            let value: &[u8] = &data[start_idx..(end_idx + 1)];
+                            let padding: Vec<u8> = vec![0; (type_size - value.len())];
+                            let padding_slice: &[u8] = &padding[..];
+                            let join = [value, padding_slice].concat();
+                            let full_value = &join[..];
+
+                            let shift = x % 8;
+                            let value = <$t>::from_le_bytes(full_value.try_into()?);
+                            Ok((end_idx, value >> shift << shift))
+                        }
+                    }
+                }
+            }
+        )*
+    };
+}
+
+generate_number_field_readers!(u8, i8, u16, i16, u32, i32, u64, i64);
+
+/// A field reader for parsing data into a field type that implements the `Lookup` trait.
+/// The `bit_size` property is used to parse a raw number value first with a methodology similar to the one
+/// defined in the documentation for NumberField, after which the raw value is passed to `Lookup::from_value`
+pub struct LookupField<T> {
+    bit_size: usize,
+    _marker: PhantomData<T>,
+}
+
+impl<T> LookupField<T> {
+    pub fn new(bit_size: usize) -> Self {
+        Self {
+            bit_size,
+            _marker: Default::default(),
+        }
+    }
+}
+
+impl<T> FieldReader for LookupField<T>
+where
+    T: NmeaEnumeratedField,
+{
+    type FieldType = T;
+
+    fn read_from_data(
+        &self,
+        data: &[u8],
+        start_idx: usize,
+    ) -> Result<(usize, Self::FieldType), NmeaParseError> {
+        let end_idx = start_idx + (self.bit_size / 8);
+        let data_slice: &[u8] = &data[start_idx..(end_idx + 1)];
+        let enum_value = match self.bit_size {
+            8 => data[start_idx] as u32,
+            16 => u16::from_le_bytes(data_slice.try_into()?) as u32,
+            32 => u32::from_le_bytes(data_slice.try_into()?),
+            x if x < 8 => {
+                let shift = 8 - x;
+                (data[start_idx] >> shift) as u32
+            }
+            x if x < 16 => {
+                let shift = 16 - x;
+                let raw_val = u16::from_le_bytes(data_slice.try_into()?);
+                (raw_val >> shift) as u32
+            }
+            x if x < 32 => {
+                let shift = 32 - x;
+                let raw_val = u32::from_le_bytes(data_slice.try_into()?);
```

Comment:
I think this will not work for `bit_size in [16;23]` as (self.bit_size / 8) == 3 and `bit_size ==32` as (self.bit_size / 8) == 5